client/src/main/java/io/confluent/kafka/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -146,19 +146,56 @@ default int getId(String subject, org.apache.avro.Schema schema)
 
   int getId(String subject, ParsedSchema schema) throws IOException, RestClientException;
 
-  public List<Integer> deleteSubject(String subject) throws IOException, RestClientException;
+  public default List<Integer> deleteSubject(String subject) throws IOException,
+          RestClientException {
+    return deleteSubject(subject, false);
+  }
 
-  public List<Integer> deleteSubject(Map<String, String> requestProperties, String subject)
-      throws IOException, RestClientException;
+  public default List<Integer> deleteSubject(String subject, boolean isPermanent)
+          throws IOException, RestClientException {
+    throw new UnsupportedOperationException();
```

Comment:
Do we need this default ?

Reply:
yes, otherwise other clients in different repo which are dependent on it will fail. Thanks to @rayokota for suggesting this approach.

Reply:
same for others below.